Workflow updates

p-galligan opened the pull request:
Adds GHA support and removes TravisCI support.

p-galligan committed:
Add GHA workflows

Adds support for GHA and removes TravisCI support.

p-galligan committed:
removes travis file

Removes travis

p-galligan committed:
Update tests

Fix ruby setup

helrond commented on .github/workflows/tests.yml:
```diff
@@ -0,0 +1,42 @@
+    env:
+      NOKOGIRI_USE_SYSTEM_LIBRARIES: true
+      JEKYLL_ENV: development
+      TRAVIS_CI: true
```
Can we use a different variable that's more generic? I don't remember what this is doing in the application code but maybe it can be combined with the JEKYLL_ENV environment variable?

helrond replied:
Ah I see this now in `script/build`. I think setting `JEKYLL_ENV` to `ci` and then updating line 15 accordingly would do the trick.

p-galligan replied:
Hey, not sure if I'm following this. Do you mean updating line 15 in the test workflow? Or line 15 in the script/build (both deal with environment stuff).

p-galligan replied:
I've pushed up a revision that I think is removing the unnecessary TRAVIS_CI variable, but it does change the build script slightly

helrond reviewed (changes requested):
Just one minor thing to decrease the number of environment variables we need to pass.

p-galligan committed:
Update workflows and build

Removes unnecessary TRAVIS_CI variable, and updates script to use `ci` for non-production builds

HaSistrunk commented:
development branch protection rule should "Require a pull request before merging", right?

p-galligan commented:
> development branch protection rule should "Require a pull request before merging", right?

Yep, missed that. Fixed now.
